### PR TITLE
FST-68 Remove system-user related functionality

### PR DESCRIPTION
**Pull request by GeloPakDev1**
### Purpose
Remove system-user related functionality

### Approach
Investigation based on the functionality included from this PR("https://github.com/folio-org/folio-spring-support/pull/107/files") and dependecies which can occur in other modules

### Learning
- Investigate classes and methods from the PR mentioned above which contained in "folio-service-tools" 
- Usage of classes and methods from the "folio-spring-support" in other modules

### CODE REVIEW

File: folio-service-tools-spring-dev/src/main/java/org/folio/spring/tools/kafka/KafkaScopedExecutionService.java [hunk 1]
```diff
@@ -1,61 +0,0 @@
...
-import lombok.SneakyThrows;
-import org.folio.spring.FolioExecutionContext;
-import org.folio.spring.scope.FolioExecutionContextSetter;
-import org.folio.spring.tools.context.ExecutionContextBuilder;
-import org.springframework.messaging.MessageHeaders;
-import org.springframework.stereotype.Service;
-
-@Service
-@RequiredArgsConstructor
-public class KafkaScopedExecutionService {
```
Comment (viacheslavkol) [inline folio-service-tools-spring-dev/src/main/java/org/folio/spring/tools/kafka/KafkaScopedExecutionService.java:14]: Why was this one removed? I can't see it in folio-spring-support

File: folio-service-tools-spring-dev/src/main/java/org/folio/spring/tools/kafka/TenantIdCheckInterceptor.java [hunk 2]
```diff
@@ -1,61 +0,0 @@
...
- *spring:
- *   kafka:
- *     producer:
- *       properties:
- *         interceptor:
- *           classes: org.folio.spring.tools.kafka.TenantIdCheckInterceptor
- * </pre>
- */
-@Log4j2
-public class TenantIdCheckInterceptor implements ProducerInterceptor<String, String> {
```
Comment (viacheslavkol) [inline folio-service-tools-spring-dev/src/main/java/org/folio/spring/tools/kafka/TenantIdCheckInterceptor.java:26]: Why was this one removed? I can't see it in folio-spring-support

**Review by psmagin — CHANGES_REQUESTED** [review]
1. Update NEWS file
2. Change the version in pom.xml to next major one as this is a breaking change.
3. While you will squash ane merge this PR please follow conventional commit format described on a page: https://wiki.folio.org/display/FOLIJET/Conventional+Commit+Initiative. It's required to have in a commit message an issue number. Please reach me directly if any help with commit message needed.

File: NEWS.md [hunk 3]
```diff
@@ -1,20 +1,6 @@
-## v3.2.0 In progress
```
Comment (psmagin) [inline NEWS.md:1]: Please back it into "In progress" and v4.0.0

File: folio-service-tools-spring-dev/pom.xml [hunk 4]
```diff
@@ -3,7 +3,7 @@
   <parent>
     <artifactId>folio-service-tools</artifactId>
     <groupId>org.folio</groupId>
-    <version>3.2.0-SNAPSHOT</version>
```
Comment (psmagin) [inline folio-service-tools-spring-dev/pom.xml:6]: It should be 4.0.0-SNAPSHOT

Comment (psmagin) [reply folio-service-tools-spring-dev/pom.xml:6]: all pom.xml should be updated with this new version

**Review by psmagin — CHANGES_REQUESTED** [review]

**Review by psmagin — APPROVED** [review]

**Pull request merged**